### ISSUE TITLE
Training crash on first validation: Error in processing validate dataset path

### ISSUE THREAD

**Issue by thanhminhmr** [ultralytics/yolov5#3644]
The tool crash while training from scratch, the trace point to this line in _test.py_.
```
Traceback (most recent call last):
  File "train.py", line 541, in <module>
    train(hyp, opt, device, tb_writer)
  File "train.py", line 367, in train
    compute_loss=compute_loss)
  File "/usr/local/lib/python3.7/dist-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/content/yolov5/test.py", line 81, in test
    is_coco = data['val'].endswith('coco/val2017.txt')  # COCO dataset
AttributeError: 'list' object has no attribute 'endswith'
```
https://github.com/ultralytics/yolov5/blob/d808855f7703f12025c0a169136c624397add112/test.py#L81

This is most likely an oversight in the code:
While the dataset file allows list of train/validation data sources (for example in _coco128.yaml_), the portion of code from above doesn't accept list.
https://github.com/ultralytics/yolov5/blob/d808855f7703f12025c0a169136c624397add112/data/coco128.yaml#L12

This is the related part of the _dataset.yaml_.
```
train: [/train/images/1/, /train/images/2/]
val: [/val/images/1/]
```

I'm unfamiliar with python so I cannot create a patch, but this should be a simple fix. (By adding type check before the _endswith_ call)

**Comment by glenn-jocher** [comment]
@thanhminhmr thanks for the bug report! Yes this looks like a problem. This should be fixable by updating this line to:
```python
is_coco = type(data['val']) is str and data['val'].endswith('coco/val2017.txt')  # COCO dataset

```

If this fix works well in your use case can you please submit a PR with it to help everyone else in the future? Thank you!

**Comment by glenn-jocher** [comment]
@thanhminhmr good news 😃! Your original issue may now be fixed ✅ in PR #3646 and #5521. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
